### PR TITLE
Add system spec for home page facets

### PR DESCRIPTION
**Pull request by cpjmcquillan**
Follow up from #2805

### CODE REVIEW

File: spec/system/jobseekers_can_view_job_facet_landing_pages_spec.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    allow(VacancyFacets).to receive(:new).and_return(vacancy_facets)
+    home_page.load
+  end
+
+  describe "job roles" do
+    it "jobseekers can view landing pages" do
+      home_page.job_roles.go_to("Teacher")
+
+      expect(current_path).to eq(job_role_path("teacher"))
+      expect(jobs_page.filters.teacher).to be_checked
```
Comment (cesidio) [inline spec/system/jobseekers_can_view_job_facet_landing_pages_spec.rb:20]: love this

File: spec/system/jobseekers_can_view_job_facet_landing_pages_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    it "jobseekers can view counties landing pages" do
+      home_page.counties.go_to("Devon")
+
+      expect(current_path).to eq(location_path("Devon"))
+      expect(jobs_page.filters.location.value).to eq("Devon")
+    end
+  end
+
+  def home_page
+    @home_page ||= PageObjects::Home.new
```
Comment (csutter) [inline spec/system/jobseekers_can_view_job_facet_landing_pages_spec.rb:53]: These two would be better served as `let`s:

```ruby
let(:home_page) { PageObjects::Home.new }
let(:jobs_page) { PageObjects::Vacancy::Index.new }
```

File: spec/system/jobseekers_can_view_job_facet_landing_pages_spec.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    store.set(:subjects, { Bengali: 5 }.to_json)
+    store.set(:cities, { London: 10 }.to_json)
+    store.set(:counties, { Devon: 15 }.to_json)
+    vacancy_facets = VacancyFacets.new(store: store)
+    allow(VacancyFacets).to receive(:new).and_return(vacancy_facets)
+    home_page.load
+  end
+
+  describe "job roles" do
+    it "jobseekers can view landing pages" do
```
Comment (cesidio) [inline spec/system/jobseekers_can_view_job_facet_landing_pages_spec.rb:16]: does it read well? maybe we should read it starting with something like

`it checks the job roles filter on results page`

**Review by cesidio — APPROVED** [review]

File: spec/system/jobseekers_can_view_job_facet_landing_pages_spec.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+RSpec.describe "Jobseekers can view job facet landing pages" do
+  before do
+    store = MockRedis.new
+    store.set(:job_roles, { teacher: 1 }.to_json)
+    store.set(:subjects, { Bengali: 5 }.to_json)
+    store.set(:cities, { London: 10 }.to_json)
+    store.set(:counties, { Devon: 15 }.to_json)
+    vacancy_facets = VacancyFacets.new(store: store)
+    allow(VacancyFacets).to receive(:new).and_return(vacancy_facets)
```
Comment (csutter) [inline spec/system/jobseekers_can_view_job_facet_landing_pages_spec.rb:11]: Not one for this PR as we do this in another spec already, but I'm always a bit hesitant to stub things at this low a level in system specs.

We could think about having something like the `vcr` gem to capture Algolia requests?

Comment (cpjmcquillan) [reply spec/system/jobseekers_can_view_job_facet_landing_pages_spec.rb:11]: Yeah I just copy 🍝 -d. I'll check it out 👁️

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-2808.london.cloudapps.digital>

**Pull request merged**